src/Serialization/containers.jl

```diff
@@ -40,20 +40,43 @@ function load_internal_with_parent(s::DeserializerState,
 end
 
 ################################################################################
-# Saving and loading tuples
-function save_internal(s::SerializerState, tup::Tuple)
+# Saving and loading Tuple
+@registerSerializationType(Tuple)
+
+function save_internal(s::SerializerState, tup::T) where T <: Tuple
+    n = fieldcount(T)
     return Dict(
+        :field_types => [encodeType(fieldtype(T, i)) for i in 1:n],
```

Comment:
I wonder how this will work in practice in various edge cases? Have you tested with e.g. tuples that contain `Vector{Int}` or `Vector{Any}` or other tuples? Maybe it trivially works, it's just not so clear to me. (Also other juxtapositions: a vector containing tuples?)

Reply:
(This is not a request for this PR. Could be done in the future, or not... just thinking out loud.)

Reply:
so it does work, i think this was actually the first example i tried
```
julia> v = ([1, 2, 3], [2, 3])
([1, 2, 3], [2, 3])

julia> save("/home/antony/test_file.json", v)
555

julia> loaded = load("/home/antony/test_file.json")
([1, 2, 3], [2, 3])
```

Reply:
Does loading Vector{Any} make sense? this would have to be implemented since
```
julia> v = ([1, 2, 3], Vector{Any}([2, 3]))
([1, 2, 3], Any[2, 3])

julia> typeof(v)
Tuple{Vector{Int64}, Vector{Any}}

julia> save("/home/antony/test_file.json", v)
555

julia> loaded = load("/home/antony/test_file.json")
([1, 2, 3], [2, 3])

julia> typeof(loaded)
Tuple{Vector{Int64}, Vector{Int64}}
```

Reply:
seems to trivially work,
```
julia> v = [(1, 2), (1, 3)]
2-element Vector{Tuple{Int64, Int64}}:
 (1, 2)
 (1, 3)

julia> save("/home/antony/test_file.json", v)
566

julia> loaded = load("/home/antony/test_file.json")
2-element Vector{Tuple{Int64, Int64}}:
 (1, 2)
 (1, 3)

julia> v = ((1, 2), (1, 3))
((1, 2), (1, 3))

julia> save("/home/antony/test_file.json", v)
598

julia> loaded = load("/home/antony/test_file.json")
((1, 2), (1, 3))
```